Jenkinsfile

```diff
@@ -1,29 +1,170 @@
-@Library('pipeline-library@ui-pipeline-legacy') _
+import groovy.json.JsonBuilder
+
+@Library('pipeline-library') _
+
+def MAIN_BRANCH = 'master'
+def DEVELOP_BRANCH = 'develop'
+
+def isMain = {
+  env.BRANCH_NAME == MAIN_BRANCH
+}
+
+def isRelease = {
+  env.BRANCH_NAME.startsWith('release/')
+}
+
+def isDevelop = {
+  env.BRANCH_NAME == DEVELOP_BRANCH
+}
+
+def getBuildType = {
+  isMain()
+    ? 'MAINLINE'
+    : 'FEATURE'
+}
 
 webappPipeline {
-    slaveLabel = 'dev_v2'
-    useArtifactoryRepo = false
     projectName = 'genesys-cloud-client-logger'
-    manifest = directoryManifest('dist')
-    buildType = { (env.BRANCH_NAME == 'master' || env.BRANCH_NAME.startsWith('release/')) ? 'MAINLINE' : 'FEATURE' }
-    publishPackage = { 'prod' }
+    team = 'Genesys Client Media (WebRTC)'
+    mailer = 'purecloud-client-media@genesys.com'
+    chatGroupId = '763fcc91-e530-4ed7-b318-03f525a077f6'
+
+    nodeVersion = '14.x'
+    buildType = getBuildType
+    manifest = customManifest('./dist') {
+        readJSON(file: 'dist/manifest.json')
+    }
+
+    testJob = 'no-tests'
+
+    ciTests = {
+      sh("""
+        npm i -g npm@7
+        npm ci
+        npm run test
+      """)
+    }
 
-    buildStep = {
+    buildStep = {cdnUrl ->
         sh("""
-          npm install npm@7 -g && \
-          npm ci && \
-          npm test && \
-          npm run build
+            echo 'CDN_URL ${cdnUrl}'
+            npm --versions
+            npm run build
         """)
     }
 
-    cmConfig = {
-        return [
-            managerEmail: 'purecloud-client-media@genesys.com',
-            rollbackPlan: 'Patch version with fix',
-            qaId: '5d41d9195ca9700dac0ef53a'
-        ]
-    }
+    deployConfig = [
+      dev : 'always',
+      test : 'always',
+      prod : 'always',
+      'fedramp-use2-core': 'always'
+    ]
+
+    onSuccess = {
```

Comment:
This function is big and ugly. This is kind of the POC run of this. Once we know this works, I will pull this out and clean it up for reuse for our other repos.